homeassistant/components/adaptive_lighting/switch.py

```diff
@@ -0,0 +1,487 @@
+        self._name = data[CONF_NAME]
+        self._lights = data[CONF_LIGHTS]
+        self._disable_brightness_adjust = data[CONF_DISABLE_BRIGHTNESS_ADJUST]
+        self._disable_entity = data[CONF_DISABLE_ENTITY]
+        self._disable_state = data[CONF_DISABLE_STATE]
+        self._initial_transition = data[CONF_INITIAL_TRANSITION]
+        self._interval = data[CONF_INTERVAL]
+        self._max_brightness = data[CONF_MAX_BRIGHTNESS]
+        self._max_color_temp = data[CONF_MAX_COLOR_TEMP]
+        self._min_brightness = data[CONF_MIN_BRIGHTNESS]
+        self._min_color_temp = data[CONF_MIN_COLOR_TEMP]
+        self._only_once = data[CONF_ONLY_ONCE]
+        self._sleep_brightness = data[CONF_SLEEP_BRIGHTNESS]
+        self._sleep_color_temp = data[CONF_SLEEP_COLOR_TEMP]
+        self._sleep_entity = data[CONF_SLEEP_ENTITY]
+        self._sleep_state = data[CONF_SLEEP_STATE]
+        self._sunrise_offset = data[CONF_SUNRISE_OFFSET]
+        self._sunrise_time = data[CONF_SUNRISE_TIME]
+        self._sunset_offset = data[CONF_SUNSET_OFFSET]
+        self._sunset_time = data[CONF_SUNSET_TIME]
+        self._transition = data[CONF_TRANSITION]
+
+        # Set other attributes
+        self._icon = ICON
+        self._entity_id = f"switch.{DOMAIN}_{slugify(self._name)}"
+
+        # Initialize attributes that will be set in self._update_attrs
+        self._percent = None
+        self._brightness = None
+        self._color_temp_kelvin = None
+        self._color_temp_mired = None
+        self._rgb_color = None
+        self._xy_color = None
+        self._hs_color = None
+
+        # Set and unset tracker in async_turn_on and async_turn_off
+        self.unsub_tracker = None
+        _LOGGER.debug(
+            "Setting up with '%s',"
+            " config_entry.data: '%s',"
+            " config_entry.options: '%s', converted to '%s'.",
+            self._lights,
+            config_entry.data,
+            config_entry.options,
+            data,
+        )
+
+    @property
+    def entity_id(self):
+        """Return the entity ID of the switch."""
+        return self._entity_id
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if adaptive lighting is on."""
+        return self.unsub_tracker is not None
+
+    def _supported_features(self, light):
+        state = self.hass.states.get(light)
+        supported_features = state.attributes["supported_features"]
+        return {
+            key for key, value in _SUPPORT_OPTS.items() if supported_features & value
+        }
+
+    def _unpack_light_groups(self, lights):
+        all_lights = []
+        for light in lights:
+            state = self.hass.states.get(light)
+            if state is None:
+                _LOGGER.debug("State of %s is None", light)
+                # TODO: make sure that the lights are loaded when doing this
```

Comment:
@frenck, I try to solve that issue here but not the TODO. Sometimes it happens that the lights aren't loaded yet when this component is set up.

I couldn't find the proper syntax to make the setting up of this component wait until light groups have been loaded. Do you know how to do that?

Reply:
Don't initialize it until HA starts.

Reply:
What is the syntax to do that?

Reply:
```
            if hass.state == CoreState.running:
                _unpack_light_groups()
            else:
                hass.bus.listen_once(
                    EVENT_HOMEASSISTANT_START, _unpack_light_groups
                )
```
The first branch is to handle the user adding the integration when already started, the second handles being loaded while starting. Also, I think it might be better to use `async_listen_once`?

Reply:
Great, thanks a lot! I have fixed this in e83741a.